File: credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int digits[16];
int arr1[16];
int arr2[16];
int x = 0;

int main(void)
{
    printf("CC Number: ");
    long n = get_long_long();
    long m = n;
//separates cc to digits
    for(int i = 0; i < 16; i++)
        {
            digits[i] = n % 10;
            n = n / 10;
        
        }

//separates digits to just every other 1    
    for(int i = 1; i < 16; i = i + 2)
    {
        arr1[x] = digits[i];
        x++;
        
    }
    
    x = 0;
    
//separates digits to just every other 2    
    for(int i = 0; i < 16; i = i + 2)
    {
        arr2[x] = digits[i];
        x++;
        
    }
    
//multiplies each in arr1 by 2     
    for(int i = 0; i < 16; i++)
    {
        arr1[i] = arr1[i] * 2;
        
    }
    
    int sum = 0;
    
// sums products' digits    
    for(int i = 0; i < 16; i++)
    {
        sum = sum + (arr1[i] % 10);
        arr1[i] = arr1[i] / 10;
        sum = sum + (arr1[i] % 10);
        
    }
    
// sums with arr2    
    for(int i = 0; i < 16; i++)
    {
        sum = sum + arr2[i];
    }

int count = 0;
n=m;
while (n != 0)
{
    n /= 10;
    count++;
}

n=m;

if((sum % 10) == 0 )
{
    //printf("%i", count)
    if (count == 15)
    {
        printf("AMEX\n");
    }
    
    else if (count == 13)
    {
        printf("VISA\n");
    }
    
    else if (count == 16)
    {
        long e = pow(10,15);
        
       // printf("%ld", e);

        if (((n / e) % 10) == 4)
        {
            printf("VISA\n");
        }
        
        e = pow(10,14);

        
        if ((n / e) == 51 || (n / e) == 52 || (n / e) == 53 || (n / e) == 54 || (n / e) == 55)
        {
            printf("MASTERCARD\n");
        }
    }
    
    else
    {
        printf("INVALID\n");
    }
    
    
}
//printf("%i", sum);
if (sum % 10 != 0)
{
    printf("INVALID\n");
}

return 0;

}

File: credit.py
import cs50
import sys

def main ():
    print ("CC Number: ", end= "")
    n = cs50.get_int()
    m=n

    #gets digits in n
    count = 0
    while True:
        n /= 10
        count += 1
        if n < 10:
            count += 1
            break

    n=m
    #separates cc to array of ints
    digits = []
    arr1 = []
    arr2 = []

    for i in range(count):
        digits.append(int (n % 10))
        n /= 10

    #separates digits to every other 1
    for i in range(1, count, 2):
        arr1.append(digits[i])

    #separates digits to every other 2
    for i in range(0, count, 2):
        arr2.append(digits[i])

    for i in range(len(arr1)):
        arr1[i] = arr1[i] * 2

    s = 0

    #sums products' digits
    for i in range(len(arr1)):
        s = s + (int (arr1[i] % 10))
        arr1[i] = arr1[i] / 10
        s = s + (int (arr1[i] % 10))

    #sums with arr2
    for i in range(len(arr2)):
        s = s + arr2[i]

    digits.reverse()
    if int (s % 10) == 0:
        if count == 15:
            print("AMEX")

        elif count == 13:
            print("VISA")

        elif count == 16:

            a = str(digits[0]) + str(digits[1])

            if digits[0] == 4:
                print("VISA")

            elif a in ['51', '52', '53', '54', '55']:
                print("MASTERCARD")
            else:
                print("INVALID")
        else:
            print("INVALID")
    else:
        print("INVALID")

if __name__ == "__main__":
    main()